lib/Dialect/FIRRTL/Transforms/WireDFT.cpp

```diff
@@ -130,54 +134,69 @@ void WireDFTPass::runOnOperation() {
-                            module.getArgument(i), anno.getFieldID());
-      enableModule = module;
-      return true;
-    });
-    if (error)
-      return signalPassFailure();
 
-    // Walk the module body looking for any operation marked as the DFT enable.
-    auto walkResult = module->walk([&](Operation *op) {
-      AnnotationSet::removeAnnotations(op, [&](Annotation anno) {
-        // If we have already encountered an error or this is not the right
-        // annotation kind, continue.
-        if (error || !anno.isClass(dftTestModeEnableAnnoClass))
-          return false;
-        if (enableSignal) {
+    auto handleAnnotation = [&](Value value, Annotation anno,
+                                StringRef annoName, Value &signal,
+                                auto &signalModule) {
+      if (anno.isClass(annoName)) {
+        if (signal) {
           auto diag =
-              op->emitError("more than one thing marked as a DFT enable");
-          diag.attachNote(enableSignal.getLoc()) << "first thing defined here";
+              emitError(value.getLoc(), "more than one thing marked as ")
+              << annoName;
+          diag.attachNote(signal.getLoc()) << "first thing defined here";
           error = true;
           return false;
         }
+
         // Grab the enable value and remove the annotation.
-        enableSignal = getValueByFieldID(
-            ImplicitLocOpBuilder::atBlockEnd(op->getLoc(), op->getBlock()),
-            op->getResult(0), anno.getFieldID());
-        enableModule = module;
+        auto builder = ImplicitLocOpBuilder::atBlockEnd(value.getLoc(),
+                                                        value.getParentBlock());
+        builder.setInsertionPointAfterValue(value);
+        signal = getValueByFieldID(builder, value, anno.getFieldID());
+        signalModule = module;
         return true;
+      }
+      return false;
+    };
+
+    // See if this module has any port marked as the DFT enable.
+    AnnotationSet::removePortAnnotations(
+        module, [&](unsigned i, Annotation anno) {
+          if (!error && handleAnnotation(module.getArgument(i), anno,
+                                         dftTestModeEnableAnnoClass,
+                                         enableSignal, enableModule))
+            return true;
+
+          if (!error &&
+              handleAnnotation(module.getArgument(i), anno,
+                               dftClockDividerBypassAnnoClass,
+                               clockDivBypassSignal, clockDivBypassModule))
+            return true;
+
+          return false;
+        });
+    if (error)
+      return signalPassFailure();
+
+    // Walk the module body looking for any operation marked as the
+    // DFT enable.
+    auto walkResult = module->walk([&](Operation *op) {
+      AnnotationSet::removeAnnotations(op, [&](Annotation anno) {
+        if (op->getNumResults() == 0)
+          return false;
```

Comment:
Thanks for fixing this, I was wondering about this. I started wondering if any operation with no results can have annotations on them. Just ignoring these operations makes more sense.

I guess it's also not possible to mark the output port of an instance op as the signal, just due to how annotations work. (or is it?)

Reply:
> Thanks for fixing this, I was wondering about this. I started wondering if any operation with no results can have annotations on them. Just ignoring these operations makes more sense.

"Which result(s) are named by an annotation?", indeed.  For InnerSymbol operations can answer which result is named (if any), not sure what would make sense here but for now leaving as-is.

> I guess it's also not possible to mark the output port of an instance op as the signal, just due to how annotations work. (or is it?)

I don't think so, they're moved to the module IIRC.  And if on an external module WireDFT doesn't notice/support it (for either enable or bypass).  I don't think we have annotations on instances themselves at all (output port or no).